File: p12.cpp
#include <iostream>
#include <cmath>

int divisors(int n)
{
    int ndiv = 0;
    int root = (int) sqrt(n);
    
    for (int i = 1; i <= root; ++i) {
        if (0 == n%i) {
            ndiv += 2;
        }
    }

    if (root*root == n) {
        ndiv -= 1;
    }

    return ndiv;
}

int main()
{
    int number = 0;
    int i = 1;
    while (divisors(number) < 500) {
        number += i;
        i += 1;
    }

    std::cout << number << std::endl;
}

File: p12.py
import math

def ndivisors(n):
    nod = 0
    root = int(math.sqrt(n))

    for i in range(1, root+1):
        if n%i == 0:
            nod += 2
    if (root**2 == n):
        nod -= 1
    return nod

number = 0
i = 1
while ndivisors(number) < 500:
    number += i
    i += 1

print(number)
